### DISCUSSION TITLE
Question sur la doc Array

### DISCUSSION THREAD

**Discussion by jojojojojoj5564656465465** [bloodyowl/boxed#87] [General]
C'est bon ce code ?

Array.filterMap([1, 2, 3], (x) =>
  isEven(x) === 0 ? Option.Some(x) : Option.None(),
); // [2]

Je comprends pas le === 0

**Comment by bloodyowl** [comment]
c'était une erreur dans l'exemple, c'est corrigé !

**Comment by jojojojojoj5564656465465** [comment]
je crois qu'il y a une autre erreur dans la doc :   =>Some(None) || some() or None

It can be useful to distinguish values between each other: you can represent **Some(None)** with options, whereas undefined or null replace the value they intend to make optional.

**Reply by bloodyowl** [reply]
je comprends pas ce que tu veux dire

**Reply by jojojojojoj5564656465465** [reply]
Je pensais que tu voulais dire : 

It can be useful to distinguish values between each other: you can represent **Some(Value) or None** with options, whereas undefined or null replace the value they intend to make optional.

De ce que j'ai compris, Some c'est pour une valeur alors que None c'est une "erreur" donc je ne comprends pas comment il peut y avoir une erreur dans une valeur Some()

**Reply by bloodyowl** [reply]
None c'est pour représenter l'absence de valeur, mais tu peux imbriquer les options:

https://swan-io.github.io/boxed/core-concepts
